docs/16-errors-frames-etc.markdown

```diff
@@ -0,0 +1,246 @@
+returned without error has a `FUNCT` of `TOPLEVEL`. If called with no
+arguments, `FRAME` returns the topmost `FRAME` used in an application
+of `ERROR` or `LISTEN`, which was bound by the interpreter to the
+`ATOM` `LERR\ I-INTERRUPTS` ("last error").
+
+16.3.4. Examples
+
+Say you have gotten an error. You can now type at `ERROR`'s `LISTEN`
+loop and get things `EVAL`ed. For example,
+
+```
+<FUNCT <FRAME>>$
+ERROR
+<FUNCT <FRAME <FRAME>>>$
+the-name-of-the-Subroutine-which-called-ERROR:atom
+<ARGS <FRAME <FRAME>>>$
+the-arguments-to-the-Subroutine-which-called-ERROR:tuple
+```
+
+16.4. ERRET
+
+    <ERRET any frame>
+
+This `SUBR` ("error return") (1) causes the control stack to be
+stripped down to the level of *frame*, and (2) **then** returns *any*.
+The net result is that the application which generated *frame* is
+forced to return *any*. Additional side effects that would have
+happened in the absence of an error may not have happened.
+
+The second argument to `ERRET` is optional, by default the `FRAME` of
+the last invocation of `ERROR` or `LISTEN`.
+
+If `ERRET` is called with **no** arguments, it drops you **all** the
+way down to the **bottom** of the control stack -- **before** the
+level-1 `LISTEN` loop -- and then calls `LISTEN`. As always, `LISTEN`
+first ensures that MDL is receptive.
+
+Examples:
+
+```
+<* 3 <+ a 1>>$
+*ERROR*
+ARG-WRONG-TYPE
++
+LISTENING-AT-LEVEL 2 PROCESS 1
+<ARGS <FRAME <FRAME>>>$
+[a 1]
+<ERRET 5>$	;"This causes the + to return 5."
+15		;"finally returned by the *"
+
+
+Note that when you are in a call to `ERROR`, the most recent set of
+bindings is still in effect. This means that you can examine values of
+dummy variables while still in the error state. For example,
+
+<DEFINE F (A "AUX" (B "a string"))
+	#DECL ((VALUE) LIST (A) STRUCTURED (B) STRING)
+	(.B <REST .A 2>)	;Return this LIST. >$
+F
+<F '(1)>$
+
+*ERROR
+OUT-OF-BOUNDS
+REST
+LISTENING-AT-LEVEL 2 PROCESS 1
+.A$
+(1)
+.B$
+"a string"
+<ERRET '(5)>	; "Make the REST return (5)."$
+("a string" (5))
+
+16.5. RETRY
+
+    <RETRY frame>
+
+causes the control stack to be stripped down just beyond *frame*, and
+then causes the Subroutine call that generated *frame* to be done
+again. *frame* is optional, by default the `FRAME` of the last
+invocation of `ERROR` or `LISTEN`. `RETRY` differs from `AGAIN` in
+that (1) it is not intended to be used in programs; (2) it can retry
+any old *frame* (any Subroutine call), whereas `AGAIN` requires an
+`ACTIVATION` (`PROG` or `REPEAT` or "`ACT`"); and (3) if it retries
+the `EVAL` of a `FORM` that makes an `ACTIVATION`, it will cause
+rebinding in the argument `LIST`, thus duplicating side effects.
+
+16.6. UNWIND
+
+`UNWIND` is an `FSUBR` that takes two arguments. usually `FORM`s. It
+`EVAL`s the first one, and, if the EVAL returns normally, the value of
+the `EVAL` call is the value of `UNWIND`. If, however, during the
+`EVAL` a non-local return attempts to return below the `UNWIND FRAME`
```

Comment:
separate backticks around UNWIND and FRAME (it's the `FRAME` created by the call to `UNWIND`)